external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -164,3 +164,23 @@ def dump(self, path: str) -> None:
                         }
                     )
                 )
+
+    def get_dict_compatible_model(self) -> tf.keras.Model:
```

Comment:
This looks quite similar to some of the code here: https://github.com/ai2cm/fv3net/blob/0d539792b7ee9804e47b0cdb684e1e03682fe5d8/external/fv3fit/fv3fit/keras/adapters.py#L21

Can you re-use some of that code? The tests for `self.dict_compatible_model` are also quite long and similar to the tests here: https://github.com/ai2cm/fv3net/blob/0d539792b7ee9804e47b0cdb684e1e03682fe5d8/external/fv3fit/tests/keras/test_adapters.py#L48

If you added an `adapters.rename_inputs` function  to go along with `rename_dict_output`, then you could implement this function with a one-liner  like this:
```
rename_dict_outputs(rename_inputs(model, self.input_variables), self.output_variables))
```

At that point, there is probably no need for any special tests...mypy is probably enough to ensure that `self.dict_compatible_model` is a keras model.

Reply:
If you function works better, I'd be amenable to replacing `rename_dict_outputs` with it. So there could be a function like 
`rename_inputs_and_outputs(model, inputs, outputs)`. I just want to avoid the duplication (especially in the test suite).

Reply:
I added a `rename_dict_inputs` function, instead of moving this function into `adapters`. The function here makes an assumption that the input model has list inputs, and doesn't work for models that have already been passed to `ensure_dict_outputs`. The new function should work for either.